src/views/pages/scores.html

```diff
@@ -14,19 +14,40 @@
     <p>Showing {{scores.length}} scores.</p>
     <table class="table table-bordered table-striped">
         <thead>
-            <tr>
-                <th ng-click="doSort('index',true)">#</th>
-                <th ng-click="doSort(['stage.index','-index'],false)">Stage</th>
-                <th ng-click="doSort(['stage.index','round','-index'],false)">Round</th>
-                <th ng-click="doSort('table',false)">Table</th>
-                <th ng-click="doSort('team.number',false)">Team</th>
-                <th ng-click="doSort('team.name',false)">Name</th>
-                <th ng-click="doSort('score',true)">Score</th>
+            <tr class="clickable sortable">
+                <th class="span1" ng-click="doSort('index',true)">
+                    <span ># </span>
+                    <i  class="material-icons">{{sortIcon('index')}}</i>
+                </th>
+                <th class="span3" ng-click="doSort(['stage.index','-index'],false)">
+                    <span >Stage </span>
+                    <i class="material-icons">{{sortIcon(['stage.index','-index'])}}</i>
+                </th>
+                <th class="span2" ng-click="doSort(['stage.index','round','-index'],false)">
+                    <span >Round </span>
+                    <i class="material-icons">{{sortIcon(['stage.index','round','-index'])}}</i>
+                </th>
+                <th class="span2" ng-click="doSort('table',false)">
+                    <span >Table </span>
+                    <i class="material-icons">{{sortIcon('table')}}</i>
+                </th>
+                <th class="span2" ng-click="doSort('team.number',false)">
+                    <span >Team </span>
+                    <i class="material-icons">{{sortIcon('team.number')}}</i>
+                </th>
+                <th class="span2" ng-click="doSort('team.name',false)">
+                    <span >Name </span>
+                    <i class="material-icons">{{sortIcon('team.name')}}</i>
+                </th>
+                <th class="span2" ng-click="doSort('score',true)">
+                    <span >Score </span>
+                    <i class="material-icons">{{sortIcon('score')}}</i>
+                </th>
                 <th>Action</th>
             </tr>
         </thead>
-        <tr ng-repeat="score in scores | index | orderBy:sort:rev track by score.index" ng-class="{error: score.error, warning: score.modified && !score.error}">
-            <td>{{score.index + 1}}</td>
+        <tr ng-repeat="score in scores | orderBy:sort:rev track by $index" ng-class="{error: score.error, warning: score.modified && !score.error}">
```

Comment:
Sorting on `$index` isn't going to work, as it will always have incrementing numbers (even after the sort), whereas `score.index` will yield e.g. `3, 2, 1` if the list was somehow reverse-sorted.

If (as you say) the `index` filter is broken, it needs to be fixed instead.